src/inc/corprof.idl

```diff
@@ -2375,6 +2379,29 @@ interface ICorProfilerCallback7 : ICorProfilerCallback6
 }
 
 
+[
+    object,
+    uuid(5BED9B15-C079-4D47-BFE2-215A140C07E0),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback8 : ICorProfilerCallback7
+{
+    // This event is triggered whenever a metadata less method is jit compiled.
+    // These include various IL Stubs and LCG Methods.
+    // The goal is to provide profiler writers with enough information to identify
+    // it to users as beyond unknown code addresses.
+    // Note: FunctionID's provided here cannot be used to resolve to their metadata
+    //       tokens since LCG methods have no metadata.
+    //
+    // Documentation Note: ilHeader is only valid during the callback
+
+    HRESULT DynamicMethodJITCompilationStarted(FunctionID functionId, LPCBYTE ilHeader);
```

Comment:
ilHeader => pILHeader
If we have a size for this buffer we should add it as an argument rather than needing to have the profiler parse the buffer to infer the size.